File: src/main/kotlin/com/mraof/unstuckalchemy/block/UnstuckOre.java
package com.mraof.unstuckalchemy.block;

import net.fabricmc.fabric.api.block.FabricBlockSettings;
import net.fabricmc.fabric.api.tools.FabricToolTags;
import net.minecraft.block.Block;
import net.minecraft.block.BlockRenderLayer;
import net.minecraft.block.Material;

public class UnstuckOre extends Block {
    UnstuckOre(int miningLevel) {
        super(FabricBlockSettings.of(Material.STONE).strength(3.0F, 3.0F).breakByTool(FabricToolTags.PICKAXES, miningLevel).build());
    }

    @Override
    public BlockRenderLayer getRenderLayer() {
        return BlockRenderLayer.CUTOUT_MIPPED;
    }
}

File: common/src/main/kotlin/com/mraof/unstuckalchemy/block/UnstuckOre.kt
package com.mraof.unstuckalchemy.block

import me.shedaniel.architectury.registry.BlockProperties
import me.shedaniel.architectury.registry.ToolType
import net.minecraft.block.Block
import net.minecraft.block.Material

class UnstuckOre internal constructor(miningLevel: Int) : Block(
    BlockProperties.of(Material.STONE).tool(ToolType.PICKAXE, miningLevel).strength(3.0f, 3.0f)
)
